packages/react-instantsearch-hooks-dom/src/widgets/__tests__/Breadcrumb.test.tsx

```diff
@@ -0,0 +1,360 @@
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+import { useHierarchicalMenu } from 'react-instantsearch-hooks';
+
+import {
+  createMultiSearchResponse,
+  createSearchClient,
+  createSingleSearchResponse,
+} from '../../../../../test/mock';
+import { InstantSearchHooksTestWrapper, wait } from '../../../../../test/utils';
+import { Breadcrumb } from '../Breadcrumb';
+
+import type { UseHierarchicalMenuProps } from 'react-instantsearch-hooks';
+
+describe('Breadcrumb', () => {
+  const hierarchicalFacets = {
+    'hierarchicalCategories.lvl0': {
+      'Cameras & Camcorders': 1369,
+    },
+    'hierarchicalCategories.lvl1': {
+      'Cameras & Camcorders > Digital Cameras': 170,
+    },
+  };
+  const hierarchicalAttributes = Object.keys(hierarchicalFacets);
+
+  const search = jest.fn((requests) =>
+    Promise.resolve(
+      createMultiSearchResponse(
+        ...requests.map(() =>
+          createSingleSearchResponse({
+            facets: hierarchicalFacets,
+          })
+        )
+      )
+    )
+  );
```

Comment:
I think you can inline this in the search client, then use `searchClient.search` when testing what the method has been called with in your tests.

Reply:
That's great, thanks for this suggestion!